app/controllers/application_controller.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  before_action :authenticate_user!
+  helper_method :current_cart
+
+  def current_cart
+    @current_cart ||= Cart.find_or_create_by(user: current_user)
```

Comment:
this line looks tricky. What if previous cart wasn't destroyed properly? Will it reuse the same cart over and over? 
Please check with requirements whether you really HAVE TO save cart. They're using browser local storage or session for this purposes usually.
Proof: https://dev.to/mrispoli24/is-using-localstorage-for-a-guest-shopping-cart-a-bad-idea-30bm
Way to go: https://medium.com/@yxp010/simple-shopping-cart-in-rails-ece6f51b27e6